client/src/App.js

```diff
@@ -1,92 +1,103 @@
+import useToken from './hooks/useToken'
 import ArtistList from './components/Artists/ArtistList'
 import SavedSongList from './components/SavedSongs/SavedSongList'
 import Navigation from './components/Navigation/Navigation'
 
 export default App
 
 function App() {
+  const { token } = useToken()
+  console.log(token)
+
   const [artistData, setArtistData] = useState([])
+
   useEffect(() => {
     setArtistData(normalizeArtists(data.tracks.items))
   }, [])
+
   const {
     toggleCurrentSongId,
     isSongPlaying,
     currentSongId,
+    stopPlayingSongById,
     stopPlayingSong,
   } = useAudio({
     artistData,
   })
+
   const { savedSongs, saveSong, deleteSavedSong } = useArtist({ artistData })
-  const [showSavedSongList, setShowSavedSongList] = useState(false)
 
   return (
-    <AppWrapper>
-      <UnmuteMessage>Don't forget to unmute your device.</UnmuteMessage>
+    <>
       <Main>
-        {showSavedSongList === false ? (
-          <ArtistList
-            artists={artistData}
-            toggleCurrentSongId={toggleCurrentSongId}
-            isSongPlaying={isSongPlaying}
-            currentSongId={currentSongId}
-            saveSong={saveSong}
-            savedSongs={savedSongs}
-          />
-        ) : (
-          ''
-        )}
-        {showSavedSongList ? (
-          <SavedSongList
-            stopPlayingSong={stopPlayingSong}
-            deleteSavedSong={deleteSavedSong}
-            savedSongs={savedSongs}
-            toggleCurrentSongId={toggleCurrentSongId}
-            isSongPlaying={isSongPlaying}
-            currentSongId={currentSongId}
-          />
+        {token ? (
+          <Switch>
+            <Route exact path="/">
+              <UnmuteMessage>
+                <p>Please unmute your device.</p>
+              </UnmuteMessage>
+              <ArtistList
+                artists={artistData}
+                toggleCurrentSongId={toggleCurrentSongId}
+                isSongPlaying={isSongPlaying}
+                currentSongId={currentSongId}
+                saveSong={saveSong}
+                savedSongs={savedSongs}
+              />
+            </Route>
+            <Route path="/favorites">
+              <SavedSongList
+                stopPlayingSongById={stopPlayingSongById}
+                deleteSavedSong={deleteSavedSong}
+                savedSongs={savedSongs}
+                toggleCurrentSongId={toggleCurrentSongId}
+                isSongPlaying={isSongPlaying}
+                currentSongId={currentSongId}
+              />
+            </Route>
+          </Switch>
         ) : (
-          ''
+          <a href="http://localhost:3001/login">Login</a>
         )}
       </Main>
-      <Footer className="footer-fixed">
-        <Navigation onClick={() => setShowSavedSongList(!showSavedSongList)}>
-          {showSavedSongList ? 'all songs' : 'saved songs'}
-        </Navigation>
+      <Footer>
+        <Navigation onClick={stopPlayingSong} />
```

Comment:
Maybe rename it to ``handleClick`` since the "real" ``onClick`` happens within the navigation component.